sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -137,13 +139,21 @@ public CompletableResultCode forceFlush() {
     return worker.forceFlush();
   }
 
+  /**
+   * Sets the {@link MeterProvider} to use to record metrics related to processed spans. By default,
+   * the {@link BatchSpanProcessor} will record metrics using the {@link GlobalMetricsProvider}.
+   * Call this method with a configured {@link MeterProvider} to enable recording metrics if the
+   * {@link GlobalMetricsProvider} is not configured..
+   */
+  @AlphaApi
```

Comment:
Why not just leave as is and only be possible to use metrics with SPI for the moment!?

Reply:
Yeah that's my comment - it means metrics needs to be set up before tracing, which could be ok for now but want to confirm that with everyone.

Reply:
I think while still alpha we are ok, we don’t want to expose this publicly in any stable api so I would not do what is proposed in this PR